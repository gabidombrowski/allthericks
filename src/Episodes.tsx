import React, { Fragment } from "react";
import { useQuery } from "@apollo/react-hooks";
import { gql } from "apollo-boost";
import {
  Box,
  Card,
  CardContent,
  CircularProgress,
  Typography
} from "@material-ui/core/";
import "./App.css";

const RICKS_EPISODES_QUERY = gql`
  {
    episodes(filter: { name: "rick" }) {
      info {
        count
      }
      results {
        name
      }
    }
  }
`;

const Episodes = () => {
  const { loading, error, data } = useQuery(RICKS_EPISODES_QUERY);

  if (loading)
    return (
      <Box justifyContent={"center"} m={3}>
        <CircularProgress />
        <Typography>Loading...</Typography>
      </Box>
    );
  if (error) return <Typography>Error :(</Typography>;
  if (!data) return <Typography>No data available...</Typography>;

  const cards = data ? (
    data.episodes.results.map((episode: any, index: any) => (
      <Card key={index} className="Card">
        <CardContent className="CardContent">
          <Typography>
            <span className="Stat">Name: </span>
            {episode.name}
          </Typography>
        </CardContent>
      </Card>
    ))
  ) : (
    <Typography>No data available...</Typography>
  );

  return (
    <Fragment>
      <Box>Total: {data.episodes.info.count}</Box>
      <Box
        display={"inline-flex"}
        flexWrap={"wrap"}
        justifyContent={"center"}
        m={3}
      >
        {cards}
      </Box>
    </Fragment>
  );
};

export default Episodes;
