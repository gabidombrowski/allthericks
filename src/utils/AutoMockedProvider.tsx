import React, { ReactNode } from 'react'
//import { gql } from 'apollo-boost'
import { ApolloProvider } from 'react-apollo'
import { ApolloClient } from 'apollo-client'
import { InMemoryCache } from 'apollo-cache-inmemory'
import { SchemaLink } from 'apollo-link-schema'
import { makeExecutableSchema, addMockFunctionsToSchema } from 'graphql-tools'
import { printSchema, buildClientSchema } from 'graphql'
import introspectionResult from './schema.json'
import { CustomResolver } from "./mergeResolvers"

const AutoMockedProvider: React.FunctionComponent<{
  children: ReactNode
  mockResolvers?: CustomResolver
}> = ({ children, mockResolvers }) => {
  const schemaSDL = printSchema(
    buildClientSchema({ __schema: introspectionResult.__schema as any }),
  )

  const schema = makeExecutableSchema({
    typeDefs: schemaSDL,
    resolverValidationOptions: {
      requireResolversForResolveType: false,
    },
  })

  addMockFunctionsToSchema({
    schema,
    mocks: mockResolvers,
  })

  const client = new ApolloClient({
    link: new SchemaLink({ schema }),
    cache: new InMemoryCache(),
  })

  //  Useful for testing:

  //   const RICKS_QUERY = gql`
  //   {
  //     characters(page: 1, filter: { name: "rick" }) {
  //       info {
  //         count
  //       }
  //       results {
  //         image
  //         name
  //         species
  //         type
  //         origin {
  //           name
  //         }
  //       }
  //     }
  //   }
  // `

  //   client.query({ query: RICKS_QUERY }).then(result => {
  //     console.log('query: ', RICKS_QUERY)
  //     console.log('result: ', result)
  //   })

  return <ApolloProvider client={client}>{children}</ApolloProvider>
}

export default AutoMockedProvider