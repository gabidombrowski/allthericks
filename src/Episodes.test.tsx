import React from "react";
import { create, act } from "react-test-renderer";
import wait from "waait";
import waitForExpect from "wait-for-expect";
import Episodes from "./Episodes";
import AutoMockedProvider from "utils/AutoMockedProvider";

const mockResolvers = {
  Query: () => ({
    episodes: () => ({
      info: () => ({
        count: 20
      })
    })
  })
};

it("renders", async () => {
  const component = create(
    <AutoMockedProvider mockResolvers={mockResolvers}>
      <Episodes />
    </AutoMockedProvider>
  );

  await act(() => wait(0));

  await waitForExpect(() => {
    expect(component).toMatchSnapshot();
  });
});
