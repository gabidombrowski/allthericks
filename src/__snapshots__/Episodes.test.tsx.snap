// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
Array [
  <div
    className="MuiBox-root MuiBox-root-2"
  >
    Total: 
    91
  </div>,
  <div
    className="MuiBox-root MuiBox-root-45"
  >
    <div
      className="MuiPaper-root MuiPaper-elevation1 MuiCard-root Card MuiPaper-rounded"
    >
      <div
        className="MuiCardContent-root CardContent"
      >
        <p
          className="MuiTypography-root MuiTypography-body1"
        >
          <span
            className="Stat"
          >
            Name: 
          </span>
          Hello World
        </p>
      </div>
    </div>
    <div
      className="MuiPaper-root MuiPaper-elevation1 MuiCard-root Card MuiPaper-rounded"
    >
      <div
        className="MuiCardContent-root CardContent"
      >
        <p
          className="MuiTypography-root MuiTypography-body1"
        >
          <span
            className="Stat"
          >
            Name: 
          </span>
          Hello World
        </p>
      </div>
    </div>
  </div>,
]
`;
